webapp/javascript/util/formatDate.js

```diff
@@ -10,32 +11,65 @@ const multiplierMapping = {
   y: "year",
 };
 
-export default function humanReadableRange(until, from) {
-  if (until === "now" && typeof from === "string") {
-    const m = from.match(/^now-(?<number>\d+)(?<multiplier>\D+)$/);
-    if (m && multiplierMapping[m.groups.multiplier]) {
-      let multiplier = multiplierMapping[m.groups.multiplier];
-      if (m.groups.number > 1) {
-        multiplier += "s";
-      }
-      const readableDateForm = {
-        range: `Last ${m.groups.number} ${multiplier}`,
-        from: moment().add(-m.groups.number, multiplier).toDate(),
-        until: moment().toDate(),
+function convertPresetsToDate(from) {
+  const { groups } = from.match(/^now-(?<number>\d+)(?<multiplier>\D+)$/);
+  const { number, multiplier } = groups;
+  let _multiplier = multiplierMapping[multiplier];
+  if (number > 1) {
+    _multiplier += "s";
+  }
+  // convert preset to a Date object
+  const fromObjectForm = moment().add(-number, _multiplier).toDate();
+
+  return { fromObjectForm, number, _multiplier };
+}
+
+function formateDate(from, until, preset = true) {
+  const { fromObjectForm, number, _multiplier } = preset
+    ? convertPresetsToDate(from)
+    : {};
+
+  /**
+   * this default values are assinged depending on the state of the parameters from and until
+   */
+
+  const _from = Math.round(moment(fromObjectForm || from * 1000));
+  const _until =
+    until !== "now" ? Math.round(moment(until * 1000)) : Math.round(moment());
+
+  let readableDateForm = {
+    range: `Last ${number} ${_multiplier}`,
+    from: _from,
+    until: _until,
+  };
+
+  if (until === "now") {
+    // check if from is not a preset
+    if (!/^now-/.test(from)) {
```

Comment:
Can probably do ```if(until === "now" && !/^now-...."

Reply:
can't do that or the range won't be the same as the default presets label
example: "Last 5 minutes"
instead, it will be date - date
example: "Jan 25, 2021 6:12 PM - Tue, Feb 2, 2021 6:12 PM"